From f45c265998097ac6c9c13dc8d59bc559b5efccfa Mon Sep 17 00:00:00 2001
From: Taku <no.one.from.minato@gmail.com>
Date: Mon, 19 Aug 2024 09:22:16 +0500
Subject: [PATCH 078/125] Added flags and melee nerf in fishing.json (#75638)

* Added flags and melee nerf in fishing.json

Added suitable flags and adjusted melee parameters of each rod and hook.

* Typo fix

* Updated legacy to_hit and made hooks not usable as a weapon

* duplicate entry fix

* Updated dps values of extended fishing rods

* Removed updated items from legaly to hit list

* Update data/mods/TEST_DATA/expected_dps_data/junk_dps.json

Reducing floating numbers as github bot suggests

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
---
 data/json/items/tool/fishing.json             | 26 +++++++++----------
 .../TEST_DATA/expected_dps_data/junk_dps.json |  4 +--
 data/mods/TEST_DATA/legacy_to_hit.json        |  2 --
 3 files changed, 15 insertions(+), 17 deletions(-)

diff --git a/data/json/items/tool/fishing.json b/data/json/items/tool/fishing.json
index 5f14057f14..258b63c25d 100644
--- a/data/json/items/tool/fishing.json
+++ b/data/json/items/tool/fishing.json
@@ -45,8 +45,7 @@
     "price_postapoc": "25 cent",
     "material": [ "steel" ],
     "symbol": ",",
-    "color": "cyan",
-    "melee_damage": { "cut": 3 }
+    "color": "cyan"
   },
   {
     "id": "fishing_hook_bone",
@@ -59,8 +58,7 @@
     "price_postapoc": "25 cent",
     "material": [ "bone" ],
     "symbol": ",",
-    "color": "white",
-    "melee_damage": { "cut": 3 }
+    "color": "white"
   },
   {
     "id": "fishing_rod_basic",
@@ -74,10 +72,10 @@
     "price_postapoc": "10 cent",
     "to_hit": { "grip": "solid", "length": "short", "surface": "any", "balance": "neutral" },
     "material": [ "wood" ],
+    "flags": [ "SHEATH_SPEAR", "FRAGILE_MELEE", "FIREWOOD" ],
     "symbol": "/",
     "color": "brown",
     "qualities": [ [ "FISHING_ROD", 1 ] ],
-    "flags": [ "SHEATH_SPEAR" ],
     "melee_damage": { "bash": 12 }
   },
   {
@@ -101,8 +99,8 @@
     "symbol": "/",
     "color": "brown",
     "qualities": [ [ "FISHING_ROD", 2 ] ],
-    "flags": [ "SHEATH_SPEAR" ],
-    "melee_damage": { "bash": 8 }
+    "flags": [ "SHEATH_SPEAR", "FRAGILE_MELEE", "REACH_ATTACK" ],
+    "melee_damage": { "bash": 6, "cut": 1 }
   },
   {
     "id": "fishing_rod_2pc_packed",
@@ -144,7 +142,7 @@
     "longest_side": "190 cm",
     "to_hit": { "grip": "solid", "length": "long", "surface": "any", "balance": "neutral" },
     "qualities": [ [ "FISHING_ROD", 2 ] ],
-    "extend": { "flags": [ "FRAGILE_MELEE" ] },
+    "extend": { "flags": [ "FRAGILE_MELEE", "SHEATH_SPEAR", "REACH_ATTACK" ] },
     "looks_like": "fishing_rod_professional",
     "use_action": {
       "menu_text": "Disassemble",
@@ -152,7 +150,7 @@
       "target": "fishing_rod_2pc_packed",
       "msg": "You take apart your two-piece fishing rod."
     },
-    "melee_damage": { "bash": 7 }
+    "melee_damage": { "bash": 5, "cut": 1 }
   },
   {
     "id": "fishing_rod_tele_packed",
@@ -172,6 +170,7 @@
       { "type": "cotton", "portion": 4 },
       { "type": "fiberglass", "portion": 4 }
     ],
+    "flags": [ "BELT_CLIP" ],
     "symbol": "/",
     "color": "dark_gray",
     "looks_like": "PR24-retracted",
@@ -193,7 +192,7 @@
     "longest_side": "210 cm",
     "to_hit": { "grip": "solid", "length": "long", "surface": "any", "balance": "neutral" },
     "qualities": [ [ "FISHING_ROD", 2 ] ],
-    "flags": [ "SHEATH_SPEAR", "FRAGILE_MELEE" ],
+    "flags": [ "SHEATH_SPEAR", "FRAGILE_MELEE", "REACH_ATTACK" ],
     "looks_like": "fishing_rod_professional",
     "use_action": {
       "menu_text": "Retract",
@@ -201,7 +200,7 @@
       "target": "fishing_rod_tele_packed",
       "msg": "You retract your telescoping fishing rod."
     },
-    "melee_damage": { "bash": 7 }
+    "melee_damage": { "bash": 5, "cut": 1 }
   },
   {
     "id": "plastichoboreel",
@@ -213,9 +212,9 @@
     "weight": "400 g",
     "volume": "400 ml",
     "longest_side": "158 mm",
+    "to_hit": { "grip": "solid", "length": "hand", "surface": "any", "balance": "neutral" },
     "price": "13 USD",
     "price_postapoc": "7 USD 50 cent",
-    "to_hit": -1,
     "material": [ "plastic" ],
     "symbol": "/",
     "color": "black",
@@ -231,10 +230,11 @@
     "weight": "500 g",
     "volume": "380 ml",
     "longest_side": "150 mm",
+    "to_hit": { "grip": "solid", "length": "hand", "surface": "any", "balance": "neutral" },
     "price": "35 USD",
     "price_postapoc": "7 USD 50 cent",
-    "to_hit": -1,
     "material": [ "wood" ],
+    "flags": [ "FIREWOOD" ],
     "symbol": "/",
     "color": "brown",
     "qualities": [ [ "FISHING_ROD", 1 ] ],
diff --git a/data/mods/TEST_DATA/expected_dps_data/junk_dps.json b/data/mods/TEST_DATA/expected_dps_data/junk_dps.json
index 41e9dd8c88..d9b4961751 100644
--- a/data/mods/TEST_DATA/expected_dps_data/junk_dps.json
+++ b/data/mods/TEST_DATA/expected_dps_data/junk_dps.json
@@ -15,6 +15,8 @@
     "type": "test_data",
     "//": "Just junk items",
     "expected_dps": {
+      "fishing_rod_tele": 4.79702,
+      "fishing_rod_2pc": 4.912611,
       "banjo": 5.53,
       "saxophone": 5.55,
       "pool_ball": 5.58,
@@ -28,12 +30,10 @@
       "flint": 5.78,
       "rock_flaking": 5.78,
       "nailboard": 5.83,
-      "fishing_rod_tele": 5.93,
       "bone_plaster": 5.98,
       "bone_fossil": 5.98,
       "grinder_blade": 6.01,
       "circsaw_blade": 6.01,
-      "fishing_rod_2pc": 6.07,
       "recoil_stock": 6.22,
       "guitar_electric": 6.3,
       "fishing_rod_tele_packed": 6.47,
diff --git a/data/mods/TEST_DATA/legacy_to_hit.json b/data/mods/TEST_DATA/legacy_to_hit.json
index d340e9ba30..00833f5ea6 100644
--- a/data/mods/TEST_DATA/legacy_to_hit.json
+++ b/data/mods/TEST_DATA/legacy_to_hit.json
@@ -587,7 +587,6 @@
       "hk417_13",
       "hk_g3",
       "hk_g80",
-      "hoboreel",
       "hoe",
       "holo_cloak",
       "homemade_demolition_charge",
@@ -867,7 +866,6 @@
       "plaguedoctor_robe",
       "plasma_rifle",
       "plastic_boat_hull",
-      "plastichoboreel",
       "platinum_grille",
       "platinum_watch",
       "plunger_futuristic",
-- 
2.45.2

